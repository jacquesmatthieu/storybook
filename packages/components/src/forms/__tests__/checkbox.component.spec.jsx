import React from 'react';
import { shallow } from 'enzyme';

import Checkbox from '../checkbox.component';

describe('Checkbox', () => {
  let props;

  beforeEach(() => {
    props = {
      name: 'foo',
    };
  });

  const getWrapper = () => shallow(<Checkbox {...props}>Bar</Checkbox>);

  it('should render with default props', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with specified props', () => {
    // Given
    props.selected = true;

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
