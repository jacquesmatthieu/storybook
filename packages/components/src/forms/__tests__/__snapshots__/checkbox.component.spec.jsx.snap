// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render with default props 1`] = `
<checkboxcomponent__LabelCheckbox
  hasLabel={true}
  htmlFor="foo"
>
  <checkboxcomponent__InputCheckbox
    id="foo"
    name="foo"
  />
  <checkboxcomponent__StyledCheckbox
    selected={false}
  />
  <typographycomponent__Typography
    variant="input"
  >
    Bar
  </typographycomponent__Typography>
</checkboxcomponent__LabelCheckbox>
`;

exports[`Checkbox should render with specified props 1`] = `
<checkboxcomponent__LabelCheckbox
  hasLabel={true}
  htmlFor="foo"
>
  <checkboxcomponent__InputCheckbox
    id="foo"
    name="foo"
  />
  <checkboxcomponent__StyledCheckbox
    selected={true}
  />
  <typographycomponent__Typography
    variant="input"
  >
    Bar
  </typographycomponent__Typography>
</checkboxcomponent__LabelCheckbox>
`;
